src/udev/udevd.c

```diff
@@ -1293,17 +1293,18 @@ static int on_inotify(sd_event_source *s, int fd, uint32_t revents, void *userda
                 _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
                 const char *devnode;
 
-                if (udev_watch_lookup(e->wd, &dev) <= 0)
+                r = device_new_from_watch_handle(&dev, e->wd);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to create sd_device object from watch handle, ignoring: %m");
                         continue;
+                }
 
                 if (sd_device_get_devname(dev, &devnode) < 0)
                         continue;
 
                 log_device_debug(dev, "Inotify event: %x for %s", e->mask, devnode);
                 if (e->mask & IN_CLOSE_WRITE)
-                        synthesize_change(dev);
-                else if (e->mask & IN_IGNORED)
```

Comment:
hmm, why is the IN_IGNORED handling dropped?

Reply:
@yuwata 
This is also my question. 

Actually, before verified this solution, I have verified that delete path_id and path_wd both, or delete anything can solve the problem. And I think this is the real problem. As follow:
```
static void device_remove_watch_handle(sd_device *device) {
        const char *id;
        int wd;

        assert(device);

        /* First, remove the symlink from handle to device id. */
        wd = device_get_watch_handle(device);
        if (wd >= 0) {
                char path_wd[STRLEN("/run/udev/watch/") + DECIMAL_STR_MAX(int)];

                xsprintf(path_wd, "/run/udev/watch/%d", wd);
                if (access(path_wd, F_OK) < 0)
                        return;
                
                (void) unlink(path_wd);
        } else 
                return;

        /* Next, remove the symlink from device id to handle. */
        if (device_get_id_filename(device, &id) >= 0) {
                const char *path_id;

                path_id = strjoina("/run/udev/watch/", id);
                (void) unlink(path_id);
        }

        device->watch_handle = -1;
}
```

Reply:
When `IN_IGNORED` is triggered, the corresponding 'remove' uevent for the device will be triggered, and during the process of the uevent, `udev_watch_end()` is called. So, it is not necessary to handle the flag here. Moreover, when both `udev_watch_end()` is called by both the main and worker process, then `device_remove_watch_handle()` in `udev_watch_end()` may cause some race. With this commit, `udev_watch_begin()` and `udev_watch_end()` are only called by workers. So, there should be no race.

I added short comment about that in the code.